framework/core/hpp_instance.cpp

```diff
@@ -347,6 +350,19 @@ HPPInstance::HPPInstance(const std::string                            &applicati
 	}
 #endif
 
+#if defined(VK_EXT_layer_settings)
+	vk::LayerSettingsCreateInfoEXT layerSettingsCreateInfo;
+
+	// If layer settings extension enabled by sample, then activate layer settings during instance creation
+	if (std::find(enabled_extensions.begin(), enabled_extensions.end(), VK_EXT_LAYER_SETTINGS_EXTENSION_NAME) != enabled_extensions.end())
+	{
+		layerSettingsCreateInfo.settingCount = uint32_t(required_layer_settings.size());
```

Comment:
First of all, thank you @asuessenbach for all your review feedback as well as the patch above.  This is very useful input and I see now this is required to remain consistent with your framework.  I apologize for not considering the unified bindings issue in my original submission and you are absolutely correct in catching this.  The only change I will make is the addition of a `#if defined(VK_EXT_layer_settings)` guard to the definition of `LayerSettingType`.  I don't want compilation to fail with earlier SDK versions that do no support layer settings.

Reply:
I would like to ask @asuessenbach one additional question before I resolve the final issues of this PR.  Is the `#if defined(VK_EXT_layer_settings)` guard truly necessary in the context of this project?  I placed these in the code to prevent build problems if compiled against an older SDK.  However, will that ever occur with this project?  Since Vulkan-Samples uses a fixed version of Vulkan Headers (currently 277) where  `VK_EXT_layer_settings` is defined, do we have to worry?  As you noticed, I recently added a runtime check for the layer settings extention in _shader_debugprintf_, so that case is covered off.  The other layer settings case is for the _descriptor_indexing_ sample which is Apple/MoltenVK-specific and things are rapidly evolving there regarding the use of Metal Argument buffers and default support for bindless - thus I am less concerned about it.

Can you please provide your advice on this?  Removing the `#if defined(VK_EXT_layer_settings)` conditionals would make the code cleaner if there is no risk of build failures.  Or are there cases where older versions of Vulkan-Headers will be used?

Reply:
Yeah, I'm totally with @asuessenbach on this. We never guard against an extension, and we wanted to add `VK_EXT_layer_settings` support anyway (See #863).

And we also don't need backwards compat with old SDKs. Actually the samples work fine without.

Reply:
Thanks for your quick feedback @asuessenbach and @SaschaWillems.  I will remove them.